include/Dialect/RNS/IR/RNSTypeInterfaces.td

```diff
@@ -0,0 +1,33 @@
+#ifndef INCLUDE_DIALECT_RNS_IR_RNSTYPEINTERFACES_TD_
+#define INCLUDE_DIALECT_RNS_IR_RNSTYPEINTERFACES_TD_
+
+include "mlir/IR/BuiltinTypeInterfaces.td"
+
+def RNSBasisTypeInterface : TypeInterface<"RNSBasisTypeInterface"> {
+  let cppNamespace = "::mlir::heir::rns";
+  let description = [{
+    This interface is required for a type to be used as a parameter
+    to an `rns` type.
+  }];
+  let methods = [
+    InterfaceMethod<
+    /*description=*/[{
+      Returns true if this type is compatible with another type in the
```

Comment:
I'm not super excited about the vague "compatible" naming here, but I'm sure we'll have a better idea of what exactly this should model after we have more examples.